src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let newSaparator = separator;
+    const newArr = this;
+    const notAdd = [null, undefined];
+
+    if (newSaparator === null) {
+      newSaparator = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
+      const cell = newArr[i];
+
+      if (notAdd.includes(cell)) {
+        if (i + 1 !== newArr.length) {
```

Comment:
```suggestion
        if (i + 1 !== newArr.length) {
```
Щоб двічі не проводити цю пееревірку, краще один раз в циклі перевіряти, чи даний елемент не є останнім (`if (i === newArr.length) {`). І якщо це останній елемент, то тоді не додавати сепаратор. У інших випадках - додавати